hera_cal/abscal.py

```diff
@@ -735,39 +787,60 @@ def global_phase_slope_logcal(model, data, antpos, wgts=None, refant=None, verbo
         red = [bl for bl in _red if bl in keys]
         if len(red) > 0:
             reds.append(red)
-
     avg_data, avg_wgts, red_keys = avg_data_across_red_bls(DataContainer({k: data[k] for k in keys}),
                                                            antpos, wgts=wgts, broadcast_wgts=False, tol=tol, reds=reds)
     avg_model, _, _ = avg_data_across_red_bls(DataContainer({k: model[k] for k in keys}),
                                               antpos, wgts=wgts, broadcast_wgts=False, tol=tol, reds=reds)
 
-    # build linear system
-    ls_data, ls_wgts = {}, {}
+    ls_data, ls_wgts, bls, pols = {}, {}, {}, {}
     for rk in red_keys:
         # build equation string
         eqn_str = '{}*Phi_ew_{} + {}*Phi_ns_{} - {}*Phi_ew_{} - {}*Phi_ns_{}'
         eqn_str = eqn_str.format(antpos[rk[0]][0], split_pol(rk[2])[0], antpos[rk[0]][1], split_pol(rk[2])[0],
                                  antpos[rk[1]][0], split_pol(rk[2])[1], antpos[rk[1]][1], split_pol(rk[2])[1])
+        bls[eqn_str] = antpos[rk[0]] - antpos[rk[1]]
+        pols[eqn_str] = rk[2]
 
         # calculate median of unflagged angle(data/model)
         # ls_weights are sum of non-binary weights
-        delta_phi = np.angle(avg_data[rk] / avg_model[rk])
+        dm_ratio = avg_data[rk] / avg_model[rk]
+        dm_ratio /= np.abs(dm_ratio)  # This gives all channels roughly equal weight, moderating the effect of RFI (as in firstcal)
         binary_flgs = np.isclose(avg_wgts[rk], 0.0)
-        delta_phi[binary_flgs] *= np.nan
-        avg_wgts[rk][np.isinf(delta_phi) + np.isnan(delta_phi)] = 0.0
-        delta_phi[np.isinf(delta_phi) + np.isnan(delta_phi)] *= np.nan
-        ls_data[eqn_str] = np.nanmedian(delta_phi[:, edge_cut:(delta_phi.shape[1] - edge_cut)], axis=1, keepdims=True)
-        ls_wgts[eqn_str] = np.sum(avg_wgts[rk][:, edge_cut:(delta_phi.shape[1] - edge_cut)], axis=1, keepdims=True)
+        dm_ratio[binary_flgs] *= np.nan
+        avg_wgts[rk][np.isinf(dm_ratio) + np.isnan(dm_ratio)] = 0.0
+        dm_ratio[np.isinf(dm_ratio) + np.isnan(dm_ratio)] *= np.nan
```

Comment:
put the two above lines before the computation of binary_flgs (i.e. flag once)

Reply:
Done.